base/dict.jl

```diff
@@ -191,9 +191,14 @@ end
 
 
 """
-    similar(d::Dict)
-
-Create an empty dictionary with the same key and value types as `d`.
+    similar(d::Dict, [element_type=eltype(e::Dict)])
+
+Create an empty dictionary with the same key and value types as `e`. The second element
+is a pair `Pair{K,V}` of key and value types and it is optional, defaulting to
+`eltype(d)`.
+Note that `similar` is useful when writing generic code that works with different
+`Associative` types. Instead, an empty dictionary with the element type of `e` is simply
+created with `Dict{keytype{e},valtype{e}}()`.
```

Comment:
OK.

Please do correct me, but `similar` for dictionaries seems to be a doing less work than `similar` for arrays, which makes an array equal in size to the given one, while `similar(d)` only cares about `eltype(d)`. My first reaction was: this is just a constructor. It was only @pabloferz's remark that made the method more meaningful.

@KristofferC : you are right.